cmake/BLTOptions.cmake

```diff
@@ -51,9 +51,14 @@ option(ENABLE_COVERAGE   "Enables code coverage support" OFF)
 
 
 ################################
-# TPL Options
+# TPL Executable Options
 ################################
-option(ENABLE_GIT       "Enables Git support" ON)
+option(ENABLE_CPPCHECK     "Enables Cppcheck support" ON)
```

Comment:
The PR description makes it seem like this is an edge-case (i.e. wanting to use a system version of these applications, but only when they are on your path)

Perhaps these should be off by default, or controlled by the presence of ``${FOO}_EXECUTABLE``.  

Specifically, I am wondering if this feature should be opt-in rather than opt-out.

Reply:
Note: I am fine with either decision (opt-in or opt-out), but wanted to raise the question.

Reply:
controlling it based on the executable path variable is exactly what this PR is trying to fix.  We spoke about this last week when i stated it was not fully qualified during the cmake step (only on the path) and then not on the path during make.

Reply:
As for opt-in vs opt-out, it doesnt cause any problems if it cant find the executable (one very small line is output) and just creates some targets.

Reply:
Sorry if I wasn't clear. 

For the opt-in part, I was suggesting that if users want the new behavior and can't (or would prefer not to) explicitly supply the executable path, they should set ``ENABLE_UNCRUSTIFY = ON`` when they configure cmake.

I am fine with leaving it as opt-out.